tools/cli/.golangci.yml

```diff
@@ -0,0 +1,135 @@
+linters-settings:
+  gocritic:
+    enabled-tags:
+      - diagnostic
+      - experimental
+      - opinionated
+      - performance
+      - style
+  govet:
+    check-shadowing: true
+    enable:
+      - fieldalignment
+
+  revive:
+    # see https://github.com/mgechev/revive#available-rules for details.
+    ignore-generated-header: true
+    severity: warning
+    rules:
+      - name: blank-imports
+      - name: context-as-argument
+      - name: context-keys-type
+      - name: dot-imports
+      - name: error-return
+      - name: error-strings
+      - name: error-naming
+      - name: errorf
+      - name: exported
+      - name: indent-error-flow
+      - name: if-return
+      - name: increment-decrement
+      - name: var-naming
+      - name: var-declaration
+      - name: package-comments
+      - name: range
+      - name: receiver-naming
+      - name: time-naming
+      - name: unexported-return
+      - name: indent-error-flow
+      - name: errorf
+      - name: empty-block
+      - name: superfluous-else
+      - name: struct-tag
+      # Too many unusued parameters, skipping this check for now
+      #- name: unused-parameter
+      - name: unreachable-code
+      - name: redefines-builtin-id
+  maligned:
+    suggest-new: true
+  misspell:
+    locale: US
+    ignore-words:
+      - cancelled
+  lll:
+    # Default is 120. '\t' is counted as 1 character.
+    # set our project to 500, as we are adding open api field description in the schema.
+    # also, for anyone using vscode, use the following configs:
+    # "rewrap.wrappingColumn": 500 ... requires the rewrap plugin
+    # "editor.rulers": [500]
+    line-length: 500
+  nestif:
+    # minimal complexity of if statements to report, 5 by default
+    min-complexity: 7
+  gomnd:
+    settings:
+      mnd:
+        # don't include the "operation" and "assign"
+        checks: [case, return]
+  funlen:
+    lines: 360
+    statements: 120
+linters:
+  disable-all: true
+  enable:
+    - dogsled
+    - errcheck
+    - funlen
+    - gocritic
+    - gofmt
+    - goimports
+    - revive
+    - gomnd
+    - goprintffuncname
+    - gosec
+    - gosimple
+    - govet
+    - ineffassign
+    - lll
+    - misspell
+    - nakedret
+    - nolintlint
+    - rowserrcheck
+    - exportloopref
+    - staticcheck
+    - stylecheck
+    - typecheck
+    - unconvert
+    - unused
+    - whitespace
+    - thelper
+    - testifylint
+    - exhaustive
+    - makezero
+    - noctx
+    - tenv
```

Comment:
let's enable `prealloc`, `predeclared` and `whitespace`
I would also recommend you run `golangci-lint linters` it would give you a full list of linters and what you have enabled and disable and descriptions on what they do and consider if there's something you think may be useful

Reply:
Thanks for sharing the command. I checked that the list contains all the recommended linters